src/Features/Core/Portable/RemoveUnnecessaryParentheses/AbstractRemoveUnnecessaryParenthesesDiagnosticAnalyzer.cs

```diff
@@ -15,11 +19,20 @@ internal abstract class AbstractRemoveUnnecessaryParenthesesDiagnosticAnalyzer<
         where TLanguageKindEnum : struct
         where TParenthesizedExpressionSyntax : SyntaxNode
     {
+        /// <summary>
+        /// Diagnostic descriptor that will not fade anything or produce a suggestion.
+        /// Can be used to only squiggle the span.
+        /// </summary>
+        private readonly DiagnosticDescriptor _unnecessaryWithoutSuggestionWithoutFadeDescriptor;
+
         protected AbstractRemoveUnnecessaryParenthesesDiagnosticAnalyzer()
             : base(IDEDiagnosticIds.RemoveUnnecessaryParenthesesDiagnosticId,
                    new LocalizableResourceString(nameof(FeaturesResources.Remove_unnecessary_parentheses), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
                    new LocalizableResourceString(nameof(FeaturesResources.Parentheses_can_be_removed), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
         {
+            _unnecessaryWithoutSuggestionWithoutFadeDescriptor = CreateDescriptorWithId(
```

Comment:
So, this descriptor does not seem to be passed to the base type and would not be part of the `SupportedDiagnostics`. It seems to work as the descriptor created in the base constructor [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/CodeStyle/AbstractCodeStyleDiagnosticAnalyzer.cs,51) seems to have the same ID. Might be better to instead use [this](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/CodeStyle/AbstractBuiltInCodeStyleDiagnosticAnalyzer.cs,18) overload of the base constructor, like we do [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs,85)?

Reply:
Thanks for the catch.  It makes sense to explicitly register it.

To use that constructor, I will additionally pull it down into [AbstractParenthesesDiagnosticAnalyzer](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/AbstractParenthesesDiagnosticAnalyzer.cs,12) since it is the parent of AbstractRemoveUnnecessaryParenthesesDiagnosticAnalyzer.  The other 2 diagnostic descriptors are created in [AbstractCodeStyleDiagnosticAnalyzer ](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/CodeStyle/AbstractCodeStyleDiagnosticAnalyzer.cs,18).  I'll need to duplicate those in AbstractRemoveUnnecessaryParenthesesDiagnosticAnalyzer

Would it make sense to move the descriptor (and initialization) to [AbstractCodeStyleDiagnosticAnalyzer ](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/CodeStyle/AbstractCodeStyleDiagnosticAnalyzer.cs,18) as well, since it offers another option similar to the other two (e.g. no fade, no suggestion, but squiggle).
e.g. `UnnecessaryWithoutSuggestionWithoutFadeDescriptor = CreateDescriptorWithId...` inside the constructor of AbstractCodeStyleDiagnosticAnalyzer

@mavasani What do you think?

Reply:
Sounds reasonable to me.